users/vieting/experiments/librispeech/librispeech_100_ctc/fairseq_finetuning/config_01_finetune.py

```diff
@@ -0,0 +1,157 @@
+        file_extension=audio_format,
+        valid_percent=valid_percent,
+    )
+    task_creation_job.rqmt["time"] = 4
+    task = task_creation_job.out_task_path
+    return task
+
+def get_fairseq_root():
+    fairseq_root = CloneGitRepositoryJob(
+        "https://github.com/facebookresearch/fairseq",
+        checkout_folder_name="fairseq",
+        commit="91c364b7ceef8032099363cb10ba19a85b050c1c").out_repository
+    #fairseq_root = SetupFairseqJob(fairseq_root, python_path).out_fairseq_root
+    fairseq_root = SetupFairseqJob(fairseq_root).out_fairseq_root
+    return fairseq_root
+
+
+def get_fairseq_args(w2v_path, corpus_names, num_gpus=1):
+    # create wav2vec manifest for training
+    task = get_task(corpus_names=corpus_names)
+
+    # Set training and model parameters
+    fairseq_args = {
+        "common": {
+            "fp16": True,
+            "log_format": "json",
+            "log_interval": 200,
+        },
+        "checkpoint": {
+            "no_epoch_checkpoints": True,
+            "best_checkpoint_metric": "wer",
+        },
+        "task": {
+            "_name": "audio_finetuning",
+            "data": task,
+            "normalize": False,
+            "labels": "ltr"
+        },
+        "dataset": {
+            "num_workers": 6,
+            "max_tokens": 3200000,  # length of tokens in one batch
+            "skip_invalid_size_inputs_valid_test": True,
+            "valid_subset": "valid",
+        },
+        "distributed_training": {
+            "distributed_world_size": num_gpus,
+            "ddp_backend": "legacy_ddp"
+        },
+        "criterion": {
+            "_name": "ctc",
+            "zero_infinity": True,
+        },
+        "optimization": {
+            "sentence_avg": True,
+            "max_update": 80000,
+            "lr": [0.00003],
+            "update_freq": [8 // num_gpus],
+        },
+        "optimizer": {
+            "_name": "adam",
+            "adam_betas": "(0.9,0.98)",
+            "adam_eps": "1e-08",
+        },
+        "lr_scheduler": {
+            "_name": "tri_stage",
+            "phase_ratio": [0.1, 0.4, 0.5],
+            "final_lr_scale": 0.05,
+        },
+        "model": {
+            "_name": "wav2vec_ctc",
+            "w2v_path": w2v_path,
+            "apply_mask": True,
+            "mask_prob": 0.65,
+            "mask_channel_prob": 0.5,
+            "mask_channel_length": 64,
+            "layerdrop": 0.1,
+            "activation_dropout": 0.1,
+            "feature_grad_mult": 0.0,
+            "freeze_finetune_updates": 0,
+        }
+    }
+    return fairseq_args
+
+
+def main():
+    prefix_name = "experiments/librispeech/librispeech_100_ctc/fairseq/"
+    # run finetuning
+    exp_name = "base"
+    num_gpus = 2
+    gpu_mem_rqmt = 24
+    corpus_names = ["train-clean-100"]
+
+    fairseq_args = get_fairseq_args(corpus_names=corpus_names, num_gpus=num_gpus, w2v_path="/work/asr3/vieting/hiwis/pletschko/fairseq/models/wav2vec_small.pt")
```

Comment:
Please use a `Path` instead of `str` for the `w2v_path` if possible. Also, this line seems too long.